### PR TITLE
[#8004] Handle edgecase for Kotlin boolean type

### PR DESCRIPTION
**Pull request by amir20001**
Fixes: #8004

trim Kotlin property names to handle edgecase for booleans

### CODE REVIEW

File: jOOQ/src/main/java/org/jooq/impl/DefaultRecordMapper.java [hunk 1]
```diff
@@ -402,7 +402,9 @@ public int compare(Constructor<E> c1, Constructor<E> c2) {
 
                     for (int i = 0; i < parameterTypes.length; i++) {
                         Reflect parameter = Reflect.on(parameters.get(i));
-                        parameterNames.add(parameter.call("getName").<String>get());
+                        // [#8004] Clean up kotlin field name for boolean types
+                        String name = parameter.call("getName").<String>get();
+                        parameterNames.add(getPropertyName(name));
```
Comment (lukaseder) [inline jOOQ/src/main/java/org/jooq/impl/DefaultRecordMapper.java]: Thank you very much for your suggestion. `Tools.getPropertyName()` extracts the property name `xyz` from getters of the form `isXyz()` and `getXyz()` as well as from setters of the form `setXyz()`. This seems to have the unwanted side effect of translating too many potential property names. Also, surely we should add a check to make sure this is about a boolean type. I assume that a property of the form `isXyz: String` does not follow the same rules in Kotlin?

Comment (amir20001) [reply jOOQ/src/main/java/org/jooq/impl/DefaultRecordMapper.java]: thanks for the feedback, i'll go ahead and make some updates to my pr

**Comment by vojtapol** [comment]
@lukaseder, what is missing from this PR at the moment?

**Comment by lukaseder** [comment]
Thanks for the ping. This looks fine to me. Will merge soon and validate with some integration tests on our side.

**Pull request merged**